libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -901,55 +1024,46 @@ internal class MapRouteLine(
         if (style.isFullyLoaded) {
-            style.getLayer(PRIMARY_ROUTE_TRAFFIC_LAYER_ID)?.setProperties(
-                lineGradient(
-                    expression
-                )
-            )
+            style.getLayer(PRIMARY_ROUTE_TRAFFIC_LAYER_ID)?.setProperties(lineGradient(expression))
         }
     }
 
     /**
-     * Updates the route line appearance from the origin point to the point indicated in
-     * @param point representing the portion of the route that has been traveled.
+     * Updates the route line appearance from the origin point to the indicated point
+     * @param point representing the current position of the puck
      */
     fun updateTraveledRouteLine(point: Point) {
-        if (vanishingPointUpdateInhibited) {
+        if (vanishingPointUpdateInhibited ||
+            System.nanoTime() - lastIndexUpdateTimeNano > MAX_ELAPSED_SINCE_INDEX_UPDATE_NANO
+        ) {
             return
         }
 
-        ifNonNull(primaryRoute, primaryRoute?.distance()) { primaryRoute, routeDistance ->
-            val lineString = getLineStringForRoute(primaryRoute)
-
-            if (lineString.coordinates().size < 2) {
-                return
-            }
-
-            val nearestPointOnLineDistance = TurfMisc.nearestPointOnLine(
-                point,
-                lineString.coordinates(),
-                TurfConstants.UNIT_METERS
-            ).getProperty("dist").asDouble
-
-            if (nearestPointOnLineDistance > ROUTE_LINE_UPDATE_MAX_DISTANCE_THRESHOLD_IN_METERS) {
-                return
-            }
-
-            val distanceRemainingFromCache: Float = distanceRemainingCache[primaryRoute]
-                ?: primaryRoute.distance().toFloat()
-            val distanceRemaining = calculatePreciseDistanceTraveledAlongLine(
-                lineString,
-                distanceRemainingFromCache,
-                point
-            )
-            val distanceTraveled = (routeDistance - distanceRemaining)
-            val percentTraveled = abs((distanceTraveled / routeDistance)).toFloat()
-
-            if (percentTraveled > MINIMUM_ROUTE_LINE_OFFSET) {
-                val expression = getExpressionAtOffset(percentTraveled)
-                hideCasingLineAtOffset(percentTraveled)
-                hideRouteLineAtOffset(percentTraveled)
-                decorateRouteLine(expression)
+        ifNonNull(
+            primaryRouteLineGranularDistances,
+            primaryRouteRemainingDistancesIndex
+        ) { granularDistances, index ->
+            val traveledIndex = granularDistances.distancesArray[index]
+            val upcomingPoint = traveledIndex.point
+
+            /**
+             * Take the remaining distance from the upcoming point on the route and extends it
+             * by the exact position of the puck.
+             */
+            val remainingDistance =
+                traveledIndex.distanceRemaining + calculateDistance(upcomingPoint, point)
```

Comment:
What about using a different naming (based on the comment) for the explaining variable? This way `updateTraveledRouteLine` will be easier to read and understand and comment will become superfluous so it won't be necessary.